js/src/test/client.test.ts

```diff
@@ -1,93 +1,98 @@
-import { describe, it, before, after } from 'node:test';
+import { describe, it, before } from 'node:test';
 import assert from 'node:assert';
 import { Identity, Transport, createTransport } from '../index.js';
 import { PROTOCOL } from '../protocol.js';
 
 describe('Transport', () => {
-  let clientIdentity: Identity;
   let serverIdentity: Identity;
 
   before(async () => {
-    clientIdentity = await Identity.generate();
     serverIdentity = await Identity.generate();
   });
 
   it('should create transport with server public key', () => {
     const transport = new Transport(
-      clientIdentity,
-      'localhost:8080',
-      serverIdentity.getPublicKey()
+      serverIdentity,
+      'localhost:8080'
     );
-    
+
     assert(transport instanceof Transport, 'Should create transport instance');
   });
 
   it('should encrypt and decrypt request', async () => {
-    const serverPublicKey = serverIdentity.getPublicKey();
     const originalBody = new TextEncoder().encode('Hello, World!');
     const request = new Request('http://localhost:8080/test', {
       method: 'POST',
       body: originalBody
     });
 
-    const encryptedRequest = await clientIdentity.encryptRequest(request, serverPublicKey);
-    
-    // Check that headers are set
-    assert(encryptedRequest.headers.get(PROTOCOL.CLIENT_PUBLIC_KEY_HEADER), 'Client public key header should be set');
+    const { request: encryptedRequest, context } = await serverIdentity.encryptRequestWithContext(request);
+
+    // Only encapsulated key header is set
```

Comment:
Ditto this comment refers to a diff change so it's confusing without the diff context